### PR TITLE
OS-96: Payment module

### PR DESCRIPTION
**Pull request by bartekkaminski**
TASK:
https://openimis.atlassian.net/browse/OP-96

### CODE REVIEW

File: IMIS/PaymentOverview.aspx.vb [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+    Dim BI As New IMIS_BI.PaymentBI
+    Dim ePaymentDetails As New IMIS_EN.tblPaymentDetail
+    Dim ePayment As New IMIS_EN.tblPayment
+    Dim dt As New DataTable
+    Protected imisgen As New IMIS_Gen
+    Private Sub FormatForm(Optional ByVal status As Integer = 0, Optional ByVal history As Boolean = False)
+
+        If status <> 4 Then
+            'BtnMatchPayment.Enabled = False
```
Comment (dragos-dobre) [inline IMIS/PaymentOverview.aspx.vb]: Please remove the commented code. Check other files for commented code also.

File: IMIS_BL/General.vb [hunk 2]
```diff
@@ -937,4 +937,61 @@ Public Function GetRejectedReasons(ByVal ReasonId As Integer) As String
...
+
+        Dim dtStatus As New DataTable
+        dtStatus.Columns.Add("StatusID", GetType(Integer))
+        dtStatus.Columns.Add("PaymenyStatusName", GetType(String))
+        If Include = True Then
+            dtStatus.Rows.Add(-1, getMessage("L_SELECTPAYMENTSTATUS"))
+        End If
+        If Not ForReport Then
+            If Include = False Then
+                'dtStatus.Rows.Add(-1, getMessage("T_FAILEDSTATUS"))
```
Comment (dragos-dobre) [inline IMIS_BL/General.vb]: Please don't delete these commented lines as they could be used when extending the payment module based on the implementation.

File: IMIS_BL/PaymentBL.vb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        Dim APIresponse As New Object
+        Dim urlToPost As String = ConfigurationManager.AppSettings("RestfullURL")
+        Dim serializer As New System.Web.Script.Serialization.JavaScriptSerializer()
+        Json = serializer.Serialize(APIObject)
+        IMISDefault.GetDefaults(eDefaults)
+        APIKey = eDefaults.APIKey
+
+        webClient.Headers("content-type") = "application/json"
+        reqString = Encoding.Default.GetBytes(Json)
+        resByte = webClient.UploadData(urlToPost + "MatchPayment?APIKey=" + APIKey, "post", reqString)
```
Comment (dragos-dobre) [inline IMIS_BL/PaymentBL.vb]: This is a call to REST API. The MatchPayment is part of V1 modules and the APIKey (the JWT) is sent through a header and not query var. The version is also to be sent. Please make the right call to MatchPayment.

File: IMIS_DAL/PaymentDAL.vb.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+Public Function MatchedPayment(ByVal PaymentID As Integer) As DataTable
```
[the remark this answers is not in the document]
Comment (bartekkaminski) [reply IMIS_DAL/PaymentDAL.vb.txt]: It is not needed. I removed

File: IMIS_BL/General.vb (hunk 2 above)
Comment (bartekkaminski) [reply IMIS_BL/General.vb]: Ok

**Pull request merged**